workflow/controller/operator.go

```diff
@@ -4134,9 +4159,19 @@ func getChildNodeIdsRetried(node *wfv1.NodeStatus, nodes wfv1.Nodes) []string {
 	childrenIds := []string{}
 	for i := 0; i < len(node.Children); i++ {
 		n := getChildNodeIndex(node, nodes, i)
-		if n == nil || n.NodeFlag == nil {
+		if n == nil {
+			continue
+		}
+
+		// fallback code here
+		if n.NodeFlag == nil {
```

Comment:
This behaviour is now different: 
Is it safe that we could be returning StepGroup etc now?

We can potentially get the same behaviour back by checking node type and only allowing for retri-able node types.

Reply:
Double checked this, this is perfectly safe given our usage of the `getChildNodeIdsAndLastRetriedNode` function.
The name of the function is however misleading and should be changed.